pay-api/src/pay_api/services/eft_service.py

```diff
@@ -121,14 +123,40 @@ def create_shortname_refund(cls, request: Dict[str, str], **kwargs) -> Dict[str,
         current_app.logger.debug(f'Starting shortname refund : {shortname_id}')
 
         refund = cls._create_refund_model(request, shortname_id, amount, comment)
-        recipients = EFTRefundEmailList.find_all_emails()
+        cls._refund_eft_credits(shortname_id, amount)
 
+        recipients = EFTRefundEmailList.find_all_emails()
         subject = f'Pending Refund Request for Short Name {shortname}'
         html_body = cls._render_email_body(shortname, amount, comment, shortname_id)
 
         send_email(recipients, subject, html_body, **kwargs)
         refund.save()
 
+    @classmethod
+    def _refund_eft_credits(cls, shortname_id: str, amount: str):
+        """Refund the amount to eft_credits table based on short_name_id."""
+        refund_amount = Decimal(amount)
+        eft_credits = EFTCreditModel.find_by_short_name_id(int(shortname_id))
```

Comment:
EFTShortnames service has a couple of functions that might be useful
`def get_eft_credit_balance(short_name_id: int) -> Decimal:`
`def get_eft_credits(short_name_id: int) -> List[EFTCreditModel]:`

I would add validation to check if the refund amount is greater than the credit balance currently and throw business exception